packages/docusaurus-theme-classic/src/theme-classic.d.ts

```diff
@@ -43,6 +43,21 @@ declare module '@theme/BlogListPaginator' {
   export default BlogListPaginator;
 }
 
+declare module '@theme/BlogSidebar' {
```

Comment:
I understand this. change but not 100% satisfied either

We should find a way to decouple theme/plugin, and yet find a way to ensure correct end-to-end type-safety between theme/plugin

Reply:
The plugin has absolutely no knowledge of `@theme/BlogSidebar`, so it should definitely be in the theme's type def.

Reply:
Oh I see

The BlogPostPage uses types from `@docusaurus/plugin-content-blog` so this looks fine/safe.

What surprised me is that it's the BlogSidebar that declares the sidebar prop, that is then imported inside BlogPostPage.

I'm not 100% which of those 2 components should "own" the type declaration (not sure if I'm clear 🤪 )